### PR TITLE
plugins/flashrom: Don't check the quirk for coreboot firmware

### PR DESCRIPTION
**Pull request by Sean-StarLabs**
coreboot can only be updated by flashrom, so don't check the quirk if the device is running coreboot.

### CODE REVIEW

**Comment by Sean-StarLabs** [comment]
@hughsie Would you be opposed to something like this? I need to add another 4 boards, which would be another 12 entries in the quirk file

**Comment by hughsie** [comment]
Back a bit, can you give me some more context? What issue are you trying to solve?

**Comment by hughsie** [comment]
>  i.e. coreboot can only be updated by flashrom so why not enable the plugin by default

Ahh, because libflashrom isn't always 100% safe! e.g. on ChromeOS we explicitly don't want flashrom unlocking stuff at startup. Although, I guess they disable the plugin at build time, so maybe it's not the end of the world. I guess I don't 100% trust flashrom not to do something super bad on random-untested laptop. @superm1 sanity check?

File: plugins/flashrom/fu-flashrom-device.c
Comment (superm1) [inline plugins/flashrom/fu-flashrom-device.c:1]: I'm fine with the change directionally, but we shouldn't have the plugin directly getting `bios_vendor`.  We should be getting that from a helper.

**Comment by hughsie** [comment]
@Sean-StarLabs can you do a `sudo fwupdtool hwids` on coreboot please? I think there's an easier fix.

**Comment by Sean-StarLabs** [comment]
Sure:
```
Computer Information
--------------------
BiosVendor: coreboot
BiosVersion: 9.04
BiosMajorRelease: 4
BiosMinorRelease: 22
FirmwareMajorRelease: 01
FirmwareMinorRelease: 19
Manufacturer: Star Labs
Family: B62-I
ProductName: StarLite
ProductSku: B62-I
EnclosureKind: 9
BaseboardManufacturer: Star Labs
BaseboardProduct: StarLite
Hardware IDs
------------
{3396c73b-40d7-58e3-9975-098208e370ac}   <- Manufacturer + Family + ProductName + ProductSku + BiosVendor + BiosVersion + BiosMajorRelease + BiosMinorRelease
{c7598e7d-569a-54f4-913d-09339861171e}   <- Manufacturer + Family + ProductName + BiosVendor + BiosVersion + BiosMajorRelease + BiosMinorRelease
{2cce54b6-1e85-5c7c-81d6-64eb011e98c6}   <- Manufacturer + ProductName + BiosVendor + BiosVersion + BiosMajorRelease + BiosMinorRelease
{34522cdc-841c-5dc7-9ea1-a80f666ff5ef}   <- Manufacturer + Family + ProductName + ProductSku + BaseboardManufacturer + BaseboardProduct
{d42d7478-4271-54a2-a7e0-8e46f0b08856}   <- Manufacturer + Family + ProductName + ProductSku
{5e45d8d7-fed9-5d8d-9b44-e97449502a4a}   <- Manufacturer + Family + ProductName
{87c6cf74-f4a9-569e-9597-a9bc5b5d0e73}   <- Manufacturer + ProductSku + BaseboardManufacturer + BaseboardProduct
{12122d1c-c383-5583-9cb7-3ba8d220913d}   <- Manufacturer + ProductSku
{6ddad544-e2d9-519d-859d-22939714cca7}   <- Manufacturer + ProductName + BaseboardManufacturer + BaseboardProduct
{9720f9aa-d864-5baa-8e0d-112bd4161e39}   <- Manufacturer + ProductName
{87c6cf74-f4a9-569e-9597-a9bc5b5d0e73}   <- Manufacturer + Family + BaseboardManufacturer + BaseboardProduct
{12122d1c-c383-5583-9cb7-3ba8d220913d}   <- Manufacturer + Family
{d17fd437-5ed8-5958-873e-100d620e3f16}   <- Manufacturer + EnclosureKind
{c3fd82b2-d529-5121-bf95-05a50136dfb2}   <- Manufacturer + BaseboardManufacturer + BaseboardProduct
{b31aa89d-cfb8-5c67-8502-ce76a9e7b580}   <- Manufacturer
```

**Comment by hughsie** [comment]
Alternate proposal in https://github.com/fwupd/fwupd/pull/6803 perhaps?

**Comment by Sean-StarLabs** [comment]
@hughsie That works. Any suggestion on how to check if it's coreboot when checking the PcrBcr?

**Comment by hughsie** [comment]
> Any suggestion on how to check if it's coreboot when checking the PcrBcr

Not immediately; but help me context switch this into my brain: what's the problem and what are our options?

**Pull request closed without merging**